src/method.jl

```diff
@@ -0,0 +1,140 @@
+"""
+    signature(m::Method) -> Dict{Symbol,Any}
+
+Finds the expression for a method's signature as broken up into its various components
+including:
+
+- `:name`: Name of the function
+- `:params`: Parametric types defined on constructors
+- `:args`: Positional arguments of the function
+- `:whereparams`: Where parameters
+
+All components listed above may not be present in the returned dictionary if they are
+not in the function definition.
+
+Limitted support for:
+- `:kwargs`: Keyword arguments of the function.
+  Only the names will be included, not the default values or type constraints.
```

Comment:
Adding this much support is exactly as hard as detecting if there are any or not.